edb/server/protocol/ai_ext.py

```diff
@@ -1174,25 +1214,69 @@ async def aconnect_sse(
 
 
 async def _start_openai_like_chat(
+    *,
     protocol: protocol.HttpProtocol,
     request: protocol.HttpRequest,
     response: protocol.HttpResponse,
     client: http.HttpClient,
     model_name: str,
     messages: list[dict[str, Any]],
     stream: bool,
+    temperature: Optional[float],
+    top_p: Optional[float],
+    max_tokens: Optional[int],
+    seed: Optional[int],
+    safe_prompt: Optional[bool],
+    logit_bias: Optional[dict[int, int]],
+    logprobs: Optional[bool],
+    user: Optional[str],
+    tools: Optional[list[dict[str, Any]]],
 ) -> None:
+    isOpenAI = "openai" in str(getattr(client, "base_url", ""))
```

Comment:
I updated the type err here too, to:
  ```
raise TypeError(
                            """prompt.custom must be a list, where each element is
                                one of the following types:\n
                            - system message: { role: 'system', content: str }\n
                            - user message: { role: 'user', content:
                                [{ type: 'text', text: str }] }\n
                            - assistant message: { role: 'assistant', content: str,
                                optional tool_calls: [{id: str, type: 'function',
                                function: { name: str, arguments: str }}] }\n
                            - tool message:
                                { role: 'tool', content: str, tool_call_id: str }"""
                        )
```
the thing is that the message can take few shapes, role and content should always be there but there can be `tool_call_id` or `tool_calls` too depending on the message role. Maybe I should add all these somehow to the if part?

Reply:
hm maybe but I dont get then why we have all this, it will be too much to provide both the provider and client, but imo I'd use one func here for `_start_openai_chat` and `_start_openai_like_chat` 
<img width="546" alt="Screenshot 2024-11-18 at 20 50 16" src="https://github.com/user-attachments/assets/fed0f0e6-e753-4677-a7d8-9177b387d17b">

Reply:
Or perhaps we can pass `provider.name` into `_start_open_ai_like_chat`. I just think it's weird to search through the client url.